analytics/src/analytics/integrations/extracts/load_opportunity_data.py

```diff
@@ -0,0 +1,79 @@
+# pylint: disable=invalid-name, line-too-long
+"""Loads opportunity tables with opportunity data from S3."""
+
+import logging
+import os
+from contextlib import ExitStack
+
+import smart_open  # type: ignore[import]
+from pydantic import Field
+from pydantic_settings import BaseSettings
+from sqlalchemy import Connection
+
+from analytics.integrations.etldb.etldb import EtlDb
+from analytics.integrations.extracts.constants import (
+    MAP_TABLES_TO_COLS,
+    OpportunityTables,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class LoadOpportunityDataFileConfig(BaseSettings):
+    """Configure S3 properties for opportunity data."""
+
+    load_opportunity_data_file_path: str | None = Field(
+        default=None,
+        alias="LOAD_OPPORTUNITY_DATA_FILE_PATH",
+    )
+
+
+def extract_copy_opportunity_data() -> None:
+    """Instantiate Etldb class and call helper funcs to truncate and insert data in one txn."""
+    etldb_conn = EtlDb()
+
+    with etldb_conn.connection() as conn, conn.begin():
+        _trancate_opportunity_table_records(conn)
+
+        _fetch_insert_opportunity_data(conn)
+
+    logger.info("Extract opportunity data completed successfully")
+
+
+def _trancate_opportunity_table_records(conn: Connection) -> None:
+    """Truncate existing records from all tables."""
+    cursor = conn.connection.cursor()
+    schema = os.environ["DB_SCHEMA"]
```

Comment:
Do you need to specify the schema? Since this sets the search_path, I think this would implicitly affect the below command (ie. you could just write that as `TRUNCATE TABLE {table} CASCADE`)

Reply:
Just out of caution. Since it will default to public schema if i leave it out.